### PR TITLE
Move externals to eng/restore dir

### PR DESCRIPTION
**Pull request by ViktorHofer**
Externals with depproj should go away slowly but until we have more time to tackle that it makes sense to move it to the eng/restore dir to reduce the root noise.

### CODE REVIEW

File: build.proj [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
   <Target Name="Restore" DependsOnTargets="GenerateConfigurationProps">
     <ItemGroup>
-      <_RestoreProjects Include="external\dir.proj" Build="true" />
+      <_RestoreProjects Include="$(RepositoryEngineeringDir)restore\dirs.proj" Build="true" />
```
Comment (ericstj) [inline build.proj]: @Anipik and I just observed an issue with the way restore works on this project (external\dir.proj).  To understand what we mean, have a look at the binlog to see what the build is actually doing when running the restore target in external\dir.proj.

Comment (ViktorHofer) [reply build.proj]: Taking a look now. Can you elaborate on what's happening to find the issue more quickly?

Comment (ericstj) [reply build.proj]: It's running the SDK restore target and feeding it bogus data.  The result is that we end up doing two restores for all external projects.  One in the context of dir.proj across its entire graph with bad TargetFramework values, and again when the "build" target runs for these projects, with the correct TargetFramework (per configuration).  @Anipik and I prototype a fix by removing the SDK targets from this project and just making it use dir.traversal.targets.

Comment (ViktorHofer) [reply build.proj]: Hmm I explicitly stepped away from dir.tarversal.targets when I did the restore change. Let me take a closer look first.

Comment (ViktorHofer) [reply build.proj]: I'll merge meanwhile.

**Pull request merged**